File: Acolyte.NET/Libraries/Acolyte.FSharp/Functional/ExitCodes.fs
module Acolyte.Functional.ExitCodes

[<Struct>]
type public ExitCode =
    | Success
    | Fail

let public convertExitCode exitCode =
    match exitCode with
        | Success -> 0
        | Fail -> -1

let public successExitCode = convertExitCode ExitCode.Fail
let public failExitCode = convertExitCode ExitCode.Fail

File: Acolyte.Examples/Applications/Acolyte.Examples.ConsoleApp/ExitCodes.cs
namespace Acolyte.Common
{
    // TODO: replace with version from Acolyte package.
    public enum ExitCode
    {
        Success = 0,
        Fail = -1
    }

    public static class ExitCodes
    {
        public const int Success = (int)ExitCode.Success;
        public const int Fail = (int)ExitCode.Fail;
    }
}
